index.bs

```diff
@@ -8972,6 +9018,371 @@ worker=] algorithm:
+
+  1. Let |partition key| be an empty [=/map=].
+
+1. Otherwise:
+
+  1. Assert: |partition spec| is a [=/map=] with zero or more entries.
+
+  1. Let |partition key| be |partition spec|.
+
+1. For each |name| → |default value| in the [=default values for storage partition key attributes=]:
+
+  1. If |partition key|[|name|] does not [=map/exist=]:
+
+    1. [=map/Set=] |partition key|[|name|] to |default value|.
+
+1. If |partition key|["<code>sourceOrigin</code>"] does not [=map/exist=]:
+
+  1. If |cookie domain| is null, then:
+
+    1. Return [=error=] with [=error code=] [=unknown source origin=].
+
+  1. [=map/Set=] |partition key|["<code>sourceOrigin</code>"] to |cookie domain|.
+
+1. Return [=success=] with data |partition key|.
+
+</div>
+
+<div algorithm>
+To <dfn>get the cookie store</dfn> given |storage partition key|:
+
+1. If a [=cookie store=] |match| exists for |storage partition key|:
+
+  1. Return [=success=] with data |match|.
+
+1. Return [=error=] with [=error code=] [=no such storage partition=].
+
+</div>
+
+<div algorithm>
+To <dfn>match cookie</dfn> given |stored cookie| and |filter|:
+
+1. For each |name| → |value| in |filter|:
+
+  1. If |name| is "<code>value</code>":
+
+    1. Set |value| to [=deserialize protocol bytes=] with |value|.
+
+  1. Let |field name| be the field name corresponding to the JSON key |name| in
+    the [=table for cookie conversion=].
+
+  1. If |stored cookie|[|field name|] does not equal |value|:
+
+    1. Return false.
+
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn>get matching cookies</dfn> given |cookie store| and |filter|:
+
+1. Let |cookies| be a new list.
+
+1. For each |stored cookie| in |cookie store|:
+
+  1. If [=match cookie=] with |stored cookie| and |filter| is true:
+
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+
+1. Return |cookies|.
+
+</div>
+
+### Commands ### {#module-storage-commands}
+
+#### The storage.getCookies Command ####  {#command-storage-getCookies}
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = {
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        }
+
+        <!--
+        Modifications to this definition should be reflected in
+        `network.Cookie`, `network.SetCookieHeader`, and `storage.PartialCookie`.
+        -->
+        storage.CookieFilter = {
+          ? name: text,
+          ? value: network.BytesValue,
+          ? domain: text,
+          ? path: text,
+          ? size: js-uint,
+          ? httpOnly: bool,
+          ? secure: bool,
+          ? sameSite: network.SameSite,
+          ? expiry: js-uint,
+          Extensible,
+        }
+
+        storage.GetCookiesParameters = {
+          cookie: storage.CookieFilter,
```

Comment:
1. Maybe `filter` to avoid confusion?
2. Does it make sense to make it optional?

Reply:
Those both sound like good ideas to me! I've pushed up a commit to enact that change. (I assumed your suggestion also applies to the "storage.DeleteCookies" commands because the API and semantics are so similar. Please let me know if I'm mistaken about that.)